### PR TITLE
[4.0] associations tooltip

### PR DESCRIPTION
**Pull request by brianteeman**
Continues the work converting the bootstrap popover to be accessible. This time for the language associations links as seen in the image

image

### CODE REVIEW

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 1]
```diff
@@ -323,16 +323,14 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+			$classes = 'badge ' . $labelClass;
 
-			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes . '" >'. $text . '</a>'
+				. '<div role="tooltip">' . $tooltip . '</div>';
 		}
 
-		HTMLHelper::_('bootstrap.popover');
```
Comment (Quy) [inline administrator/components/com_associations/Helper/AssociationsHelper.php:334]: Remove `use Joomla\CMS\HTML\HTMLHelper;`. Repeat in other files.

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 2]
```diff
@@ -323,16 +323,14 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
-			$tooltip = htmlspecialchars($title, ENT_QUOTES, 'UTF-8') . '<br><br>' . $additional;
-			$classes = 'hasPopover badge ' . $labelClass;
+			$tooltip = '<strong>' . htmlspecialchars($language->title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+				 . htmlspecialchars($title, ENT_QUOTES, 'UTF-8') . '<br><br>' . $additional;
+			$classes = 'badge ' . $labelClass;
 
-			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes . '" >'. $text . '</a>'
```
Comment (Quy) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]:
```suggestion
			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes . '">'. $text . '</a>'
```

Comment (Quy) [reply administrator/components/com_associations/Helper/AssociationsHelper.php]: Repeat in other files.

File: administrator/components/com_associations/Helper/AssociationsHelper.php (hunk 1 above)
Comment (brianteeman) [reply administrator/components/com_associations/Helper/AssociationsHelper.php:334]: Done where appropriate

File: administrator/components/com_associations/Helper/AssociationsHelper.php (hunk 2 above)
Comment (brianteeman) [reply administrator/components/com_associations/Helper/AssociationsHelper.php]: done

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 3]
```diff
@@ -323,16 +322,14 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
-			$tooltip = htmlspecialchars($title, ENT_QUOTES, 'UTF-8') . '<br><br>' . $additional;
-			$classes = 'hasPopover badge ' . $labelClass;
+			$tooltip = '<strong>' . htmlspecialchars($language->title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+				 . htmlspecialchars($title, ENT_QUOTES, 'UTF-8') . '<br><br>' . $additional;
+			$classes = 'badge ' . $labelClass;
 
-			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes . '">'. $text . '</a>'
```
Comment (Quy) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]:
```suggestion
			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes . '">' . $text . '</a>'
```

File: administrator/components/com_categories/Service/HTML/AdministratorService.php [hunk 4]
```diff
@@ -73,17 +72,17 @@ public function association($catid, $extension = 'com_content')
...
-					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes
-						. '" data-content="' . htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '" data-placement="top">'
-						. $text . '</a>';
+					$text     = $item->lang_sef ? strtoupper($item->lang_sef) : 'XX';
+					$url      = Route::_('index.php?option=com_categories&task=category.edit&id=' . (int) $item->id . '&extension=' . $extension);
+					$tooltip  = '<strong>' . htmlspecialchars($item->language_title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+						. htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8');
+					$classes  = 'badge badge-secondary';
+
+					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">'. $text . '</a>'
```
Comment (Quy) [inline administrator/components/com_categories/Service/HTML/AdministratorService.php]:
```suggestion
					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
```

File: administrator/components/com_contact/Service/HTML/AdministratorService.php [hunk 5]
```diff
@@ -78,17 +78,15 @@ public function association($contactid)
...
-					$tooltip = htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '<br>' . Text::sprintf('JCATEGORY_SPRINTF', $item->category_title);
-					$classes = 'hasPopover badge badge-secondary';
+					$tooltip = '<strong>' . htmlspecialchars($item->language_title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+						. htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '<br>' . Text::sprintf('JCATEGORY_SPRINTF', $item->category_title);
+					$classes = 'badge badge-secondary';
 
-					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">'. $text . '</a>'
```
Comment (Quy) [inline administrator/components/com_contact/Service/HTML/AdministratorService.php]:
```suggestion
					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
```

File: administrator/components/com_content/Service/HTML/AdministratorService.php [hunk 6]
```diff
@@ -80,17 +80,15 @@ public function association($articleid)
...
-					$tooltip = htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '<br>' . Text::sprintf('JCATEGORY_SPRINTF', $item->category_title);
-					$classes = 'hasPopover badge badge-secondary';
+					$tooltip = '<strong>' . htmlspecialchars($item->language_title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+						. htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '<br>' . Text::sprintf('JCATEGORY_SPRINTF', $item->category_title);
+					$classes = 'badge badge-secondary';
 
-					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">'. $text . '</a>'
```
Comment (Quy) [inline administrator/components/com_content/Service/HTML/AdministratorService.php]:
```suggestion
					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
```

File: administrator/components/com_newsfeeds/Service/HTML/AdministratorService.php [hunk 7]
```diff
@@ -77,17 +76,15 @@ public function association($newsfeedid)
...
-					$tooltip = htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '<br>' . Text::sprintf('JCATEGORY_SPRINTF', $item->category_title);
-					$classes = 'hasPopover badge badge-secondary';
+					$tooltip = '<strong>' . htmlspecialchars($item->language_title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+						. htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8') . '<br>' . Text::sprintf('JCATEGORY_SPRINTF', $item->category_title);
+					$classes = 'badge badge-secondary';
 
-					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">'. $text . '</a>'
```
Comment (Quy) [inline administrator/components/com_newsfeeds/Service/HTML/AdministratorService.php]:
```suggestion
					$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
```

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 8]
```diff
@@ -323,16 +322,14 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
-			$classes = 'hasPopover badge ' . $labelClass;
+			$tooltip = '<strong>' . htmlspecialchars($language->title, ENT_QUOTES, 'UTF-8') . '</strong><br>'
+				 . htmlspecialchars($title, ENT_QUOTES, 'UTF-8') . '<br><br>' . $additional;
+			$classes = 'badge ' . $labelClass;
 
-			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes
-						. '" data-content="' . $tooltip . '" data-placement="top">'
-						. $text . '</a>';
+			$items[$langCode]['link'] = '<a href="' . $url . '" title="' . $language->title . '" class="' . $classes . '">' . $text . '</a>'
+				. '<div role="tooltip">' . $tooltip . '</div>';
```
Comment (Quy) [inline administrator/components/com_associations/Helper/AssociationsHelper.php:330]: This does not have `id="tip"`. OK without?

**Comment by Quy** [comment]
Remove title attribute? It is overlapping the tooltip.

associationjpg

File: administrator/components/com_associations/Helper/AssociationsHelper.php (hunk 8 above)
Comment (brianteeman) [reply administrator/components/com_associations/Helper/AssociationsHelper.php:330]: Yeah it's ok

**Comment by brianteeman** [comment]
I'm aware of that. It's better to keep the title I think and hope the styling canbr resolved by the new template

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8f4e1befbc53bab065325b41d66fbd09d3c4b71f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25312">issues.joomla.org/tracker/joomla-cms/25312</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thabks

**Pull request merged**